restli-common/src/main/java/com/linkedin/restli/common/ResourceMethodIdentifierGenerator.java

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright (c) 2013 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.restli.common;
+
+/**
+ * @author dmessink
```

Comment:
This is a new concept, so I'd suggest we document a little more heavily. I suggest explaining that this generates a unique identifier for a given Rest.li method, fully qualified by its resource hierarchy